src/_pytest/mark/legacy.py

```diff
@@ -57,7 +57,15 @@ def from_item(cls, item):
         return cls(mapped_names)
 
     def __getitem__(self, subname):
-        for name in self._names:
+        """Return whether subname is included within stored names.
+
+        The string inclusion check is case-insensitive.
+
+        """
+        subname = subname.lower()
+        names = [name.lower() for name in self._names]
```

Comment:
```suggestion
        names = (name.lower() for name in self._names)
```

Reply:
For knowledge sharing purposes. Why is a tuple preferred over a list here? 😊  @nicoddemus

Reply:
Oh that's not a tuple that's a [generator expression](https://www.python.org/dev/peps/pep-0289/). 😉

Reply:
Given that we will only iterate over it, a generator expression is more memory efficient as it will avoid creating a temporary list.

Of course I doubt this will make any difference here at all, but I prefer to just follow good practices even when they won't have much/any impact than trying to evaluate every case. 👍

Reply:
> Oh that's not a tuple that's a [generator expression](https://www.python.org/dev/peps/pep-0289/). 😉

Right, hence the strikeout of tuple, and the word generator after it. 😉 

Y tho? 😊